### ISSUE TITLE
VIP: safe erc20 transfer builtin

### ISSUE THREAD

**Issue by charles-cooper** [vyperlang/vyper#2753]
## Simple Summary
add a builtin to handle non-compliant ERC20 tokens

## Motivation
I see this in the wild all the time:
```vyper
res: Bytes[32] = raw_call(_abi_encode(sender, receiver, amount), method_id=method_id("transferFrom(address,address,uint256)"))
if len(res) > 0:
    assert convert(res, bool)
```

We could do it more efficiently as a builtin.

## Specification
add a function, (name: `safeTransferFromERC20`?), which implements a safe ERC20 transfer as above.

## Dependencies
May be good to implement this as a library function once https://github.com/vyperlang/vyper/issues/2431 is implemented.

## Copyright
Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/)

**Comment by charles-cooper** [comment]
whoops, i searched for the "VIP" label and clicked the first one

**Comment by charles-cooper** [comment]
note: `default` is just a placeholder, real name of the kwarg TBD if we decide to move forward with this idea.

**Comment by charles-cooper** [comment]
closed in favor of #2812
